crates/sui/src/ptb/ptb.rs

```diff
@@ -233,21 +242,31 @@ impl PTB {
             ));
         }
 
-        let files_to_resolve = ignore_comments
+        let mut files_to_resolve = vec![];
+
+        for f in ignore_comments
             .iter()
-            .flat_map(|x| x.split_inclusive("--"))
-            .filter(|x| x.starts_with("--file"))
-            .map(|x| x.to_string().replace("--file", "").replace(" ", ""))
-            .collect::<Vec<_>>();
+            .flat_map(|x| x.split("--"))
+            .filter(|x| x.starts_with("file"))
+            .map(|x| x.to_string().replace("file", "").replace(" ", ""))
+        {
+            let mut p = PathBuf::new();
+            p.push(parent.clone());
+            p.push(f.clone());
+            let file = std::fs::canonicalize(p)
+                .map_err(|_| anyhow!("{} includes file {} which does not exist.", filename, f))?;
+            files_to_resolve.push(file);
+        }
+
         if let Some(files) = included_files.get_mut(&current_file) {
             files.extend(files_to_resolve);
         } else {
-            included_files.insert(current_file, files_to_resolve);
+            included_files.insert(file_path, files_to_resolve);
         }
 
         let edges = included_files.iter().flat_map(|(k, vs)| {
-            let vs = vs.iter().map(|v| v.as_str());
-            std::iter::repeat(k.as_str()).zip(vs)
+            let vs = vs.iter().map(|v| v.to_str().unwrap());
+            std::iter::repeat(k.to_str().unwrap()).zip(vs)
```

Comment:
Ugh, should make this nicer, but `PathBuf`'s `to_str()` function returns an `Option`...

Reply:
I know that feeling..to only other option is to call `to_string_lossy` (or similar? I forget the exact name) but I think this is fine.